infra/main.bicep

```diff
@@ -271,6 +273,46 @@ module searchRoleBackend 'core/security/role.bicep' = {
   }
 }
 
+module searchRoleOpenAi 'core/security/role.bicep' = if (!searchApiKeyEnabled) {
+  scope: searchServiceResourceGroup
+  name: 'search-role-openai'
+  params: {
+    principalId: openAi.outputs.identityPrincipalId
+    roleDefinitionId: '1407120a-92aa-4202-b7e9-c0e197c71c8f'
+    principalType: 'ServicePrincipal'
+  }
+}
+
+module searchServiceRoleOpenAi 'core/security/role.bicep' = if (!searchApiKeyEnabled) {
+  scope: searchServiceResourceGroup
+  name: 'search-service-role-openai'
+  params: {
+    principalId: openAi.outputs.identityPrincipalId
+    roleDefinitionId: '7ca78c08-252a-4471-8644-bb5ff32d4ba0'
+    principalType: 'ServicePrincipal'
+  }
+}
+
+module openAiContributorRoleSearch 'core/security/role.bicep' = if (!searchApiKeyEnabled) {
+  scope: searchServiceResourceGroup
+  name: 'openai-contributor-role-search'
+  params: {
+    principalId: searchService.outputs.identityPrincipalId
+    roleDefinitionId: 'a001fd3d-188f-4b5d-821b-7da978bf7442'
+    principalType: 'ServicePrincipal'
+  }
+}
+
+module cognitiveServicesContributorRoleSearch 'core/security/role.bicep' = if (!searchApiKeyEnabled) {
+  scope: searchServiceResourceGroup
+  name: 'cognitive-services-contributor-role-search'
+  params: {
+    principalId: searchService.outputs.identityPrincipalId
+    roleDefinitionId: '25fbc0a9-bd7c-42a3-aa1a-3b75d497ee68'
+    principalType: 'ServicePrincipal'
+  }
+}
+
```

Comment:
In outputs below we shouldn't be outputting keys.  You will get a bicep linter warning.  Best to ask the user to manually set them with in env vars at this time until we have a secure way to store them.

Reply:
Phew, glad to see the keys are out of this output finally! That's been a TODO (in my head) for a long time.

Another option btw is to provide a script that grabs the keys, which is what I've had to do for another repo which unfortunately requires keys:
https://github.com/Azure-Samples/contoso-chat/pull/33/files#diff-c8719e3915d76118aeee543ed85367b315252d46ef1d793635a82eaef96ce7b4R19

But probably okay not to do that here, since we now have the keyless option.